megatron/model/bert_model.py

```diff
@@ -206,12 +208,14 @@ def forward(self, bert_model_input, attention_mask,
             pooled_output = None
 
         if self.post_process:
-            lm_output = lm_output if self.add_binary_head else lm_output[0]
-            return post_language_model_processing(lm_output, pooled_output,
-                                                  self.lm_head, self.binary_head,
-                                                  lm_labels,
-                                                  self.shared_embedding_or_output_weight(),
-                                                  self.fp16_lm_cross_entropy)
+            if not self.add_binary_head:
+                lm_output, moe_losses = lm_output
+            lm_output = post_language_model_processing(lm_output, pooled_output,
+                                                       self.lm_head, self.binary_head,
+                                                       lm_labels,
+                                                       self.shared_embedding_or_output_weight(),
+                                                       self.fp16_lm_cross_entropy)
+            return *lm_output, moe_losses if self.return_moe_loss else lm_output
```

Comment:
Is the `*` before the first lm_output by mistake or on purpose. If on purpose, could you add a comment to document why so?

Reply:
It's intentional, I wanted to keep the structure flat, same as in GPT, and `lm_output` is a `tuple`: (`lm_loss`, `binary_logits`).
As for a comment, my opinion - it's redundant. It's not needed more that a comment describing the output of `post_language_model_processing` - the function is there in the same file (and there's no comment now).
But let me know if you feel strongly about this one - it can be easily arranged.

Reply:
btw - not a huge fan of returning tuples myself when the meaning is loaded or ambiguous (unlike, say, `divmod`).
I would prefer namedtuples or typed structs of some sort. But as long as it's consistent and, more importantly, present in upstream Megatron-LM - it's fine by me.

Reply:
Sounds good, your comment above is a good enough documentation record. Let me merge it now.